lib/recipe.rb

```diff
@@ -0,0 +1,33 @@
+class Recipe
+    attr_reader :name,
+                :ingredients_required
+
+    def initialize(name)
+        @name                 = name
+        @ingredients_required = {}
+    end
+
+    def add_ingredient(ingredient, quantity)
+        if @ingredients_required[ingredient]
+            @ingredients_required[ingredient] += quantity
+        else 
+            @ingredients_required[ingredient] = quantity
+        end
+    end
+
+    def ingredients
+        noms = []
+        @ingredients_required.each do |ingredient, quantity|
+            noms << ingredient
+        end
+        noms.flatten
```

Comment:
You're just putting the key into the noms array. Shouldn't need to flatten it. Great job with this.